Mathlib/Data/Sym/Sym2/Finsupp.lean

```diff
@@ -0,0 +1,54 @@
+/-
+Copyright (c) 2025 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+import Mathlib.Algebra.GroupWithZero.Basic
+import Mathlib.Data.Finset.Sym
+import Mathlib.Data.Finsupp.Defs
+
+/-!
+# Finitely supported functions from the symmetric square
+
+This file lifts functions `α →₀ M₀` to functions `Sym2 α →₀ M₀` by precomposing with multiplication.
+-/
+
+open Sym2
+
+variable {α M₀ : Type*} [CommMonoidWithZero M₀] {f : α →₀ M₀}
+
+namespace Sym2
+
+lemma sym2_support_eq_preimage_support_mul [NoZeroDivisors M₀] (f : α →₀ M₀) :
+    f.support.sym2 = map f ⁻¹' mul.support := by ext ⟨a, b⟩; simp
+
+lemma mem_sym2_support_of_mul_ne_zero (p : Sym2 α) (hp : mul (p.map f) ≠ 0) :
+    p ∈ f.support.sym2 := by
+  obtain ⟨a, b⟩ := p
+  simp only [map_pair_eq, mul_mk, ne_eq] at hp
+  simpa using .intro (left_ne_zero_of_mul hp) (right_ne_zero_of_mul hp)
```

Comment:
@YaelDillies, are you happy with the `Sym2` namespace being used here?

Reply:
Whoops indeed I missed that completely. I guess `Finsupp` makes the most sense as a namespace?

Reply:
@mans0954, can you change the namespace?

Reply:
@YaelDillies yes, done. Sorry for the delay - been very busy the last couple of weeks.